cli/cmd/tapRunner.go

```diff
@@ -252,77 +251,97 @@ func updateMizuTappers(ctx context.Context, kubernetesProvider *kubernetes.Provi
-				return
-			}
-		} else {
-			if err := kubernetesProvider.RemoveServicAccount(removalCtx, config.Config.MizuResourcesNamespace, mizu.ServiceAccountName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing Service Account %s in namespace %s: %v", mizu.ServiceAccountName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-				return
-			}
+	if err := kubernetesProvider.RemoveNonNamespacedResources(ctx, mizu.ClusterRoleName, mizu.ClusterRoleBindingName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing non-namespaced resources: %v", errormessage.FormatError(err)))
+	}
 
-			if err := kubernetesProvider.RemoveRole(removalCtx, config.Config.MizuResourcesNamespace, mizu.RoleName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing Role %s in namespace %s: %v", mizu.RoleName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-			}
+	if err := kubernetesProvider.RemoveServicAccount(ctx, config.Config.MizuResourcesNamespace, mizu.ServiceAccountName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing Service Account %s in namespace %s: %v", mizu.ServiceAccountName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
+	}
 
-			if err := kubernetesProvider.RemoveRoleBinding(removalCtx, config.Config.MizuResourcesNamespace, mizu.RoleBindingName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing RoleBinding %s in namespace %s: %v", mizu.RoleBindingName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-			}
-		}
+	if err := kubernetesProvider.RemoveRole(ctx, config.Config.MizuResourcesNamespace, mizu.RoleName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing Role %s in namespace %s: %v", mizu.RoleName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
 	}
 
-	if !config.Config.IsNsRestrictedMode() {
-		waitUntilNamespaceDeleted(removalCtx, cancel, kubernetesProvider)
+	if err := kubernetesProvider.RemoveRoleBinding(ctx, config.Config.MizuResourcesNamespace, mizu.RoleBindingName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing RoleBinding %s in namespace %s: %v", mizu.RoleBindingName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
 	}
 }
 
+func isErrForbiddenGet(err error) bool {
+	if status := k8serrors.APIStatus(nil); errors.As(err, &status) {
+		return strings.Contains(status.Status().Message, "cannot get resource")
```

Comment:
can't we check by status code?

Reply:
Status code is 403 for both GET and DELETE requests.
This is the jsonified status of a failed GET operation:
```json
  {
    "metadata": {},
    "status": "Failure",
    "message": "configmaps \"mizu-policy\" is forbidden: User \"user1\" cannot get resource \"configmaps\" in API group \"\" in the namespace \"user1\"",
    "reason": "Forbidden",
    "details": {
      "name": "mizu-policy",
      "kind": "configmaps"
    },
    "code": 403
  }
```
and this is the jsonified error of a failed DELETE operation:
```json
  {
    "metadata": {},
    "status": "Failure",
    "message": "clusterroles.rbac.authorization.k8s.io \"mizu-cluster-role\" is forbidden: User \"user1\" cannot get resource \"clusterroles\" in API group \"rbac.authorization.k8s.io\" at the cluster scope",
    "reason": "Forbidden",
    "details": {
      "name": "mizu-cluster-role",
      "group": "rbac.authorization.k8s.io",
      "kind": "clusterroles"
    },
    "code": 403
  }
```

The only difference is in the message.